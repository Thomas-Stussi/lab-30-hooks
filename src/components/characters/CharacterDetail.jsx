import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ characterObj }) => (
  <>
    <h3 data-testid="detail">{characterObj.name}</h3>
    <img src={characterObj.image} />
    <p>Bio: {characterObj.description}</p>
    <a href="/">
      <button>Back to Character List</button>
    </a>

  </>
);

CharacterDetail.propTypes = {
  characterObj: PropTypes.object.isRequired
};

export default CharacterDetail;
