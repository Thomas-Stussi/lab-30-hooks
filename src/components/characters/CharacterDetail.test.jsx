/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('Detail Component', () => {
  it('renders Detail component', () => {
    const character = {
      'name': 'Zenzola',
      'gender': null,
      'status': null,
      'born': null,
      'occupation': null,
      'rank': null,
      'affiliations': null,
      'portrayedby': null,
      'image': 'https://vignette.wikia.nocookie.net/x-files/images/9/97/Zenzola.jpg/revision/latest/scale-to-width-down/240?cb=20150613121059',
      'description': 'Dr. Zenzola was a physician in Newark, New Jersey who treated Craig for his injuries following his attack in the sewers by a mysterious creature. She explained to Fox Mulder that she was uncertain what had attacked Craig, but that she\'d given the patient gum to chew to help with a bad taste in his mouth as well as a treatment of antibiotics for any bacteria he may have picked up while immersed in the sewer water fighting with the creature. She also showed Mulder the bizarre bite wound on the man\'s back.',
      'categories': [
        'Doctors'
      ]
    };
    const { asFragment } = render(<CharacterDetail characterObj={character} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
