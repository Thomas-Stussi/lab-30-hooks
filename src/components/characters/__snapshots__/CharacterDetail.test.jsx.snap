// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Component renders Detail component 1`] = `
<DocumentFragment>
  <h3
    data-testid="detail"
  >
    Zenzola
  </h3>
  <img
    src="https://vignette.wikia.nocookie.net/x-files/images/9/97/Zenzola.jpg/revision/latest/scale-to-width-down/240?cb=20150613121059"
  />
  <p>
    Bio: Dr. Zenzola was a physician in Newark, New Jersey who treated Craig for his injuries following his attack in the sewers by a mysterious creature. She explained to Fox Mulder that she was uncertain what had attacked Craig, but that she'd given the patient gum to chew to help with a bad taste in his mouth as well as a treatment of antibiotics for any bacteria he may have picked up while immersed in the sewer water fighting with the creature. She also showed Mulder the bizarre bite wound on the man's back.
  </p>
  <a
    href="/"
  >
    <button>
      Back to Character List
    </button>
  </a>
</DocumentFragment>
`;
